src/main/java/htsjdk/samtools/sra/SRAAlignmentIterator.java

```diff
@@ -191,4 +196,14 @@ private void nextReference() {
 
         return referencesChunks;
     }
+
+    @Override
+    public void close() {
+        if (lastRecord != null) {
+            lastRecord.detachFromIterator();
```

Comment:
`lastRecord` object is an instance of SRALazyRecord class, which keeps a reference to NGS.AlignmentIterator for getting fields which were not loaded yet. That means that if somebody holds a reference to lastRecord outside of this class, he will also hold a reference to AlignmentIterator and that will prevent it from being garbage collected